dask_snowflake/core.py

```diff
@@ -140,12 +140,60 @@ def _fetch_query_batches(query, connection_kwargs, execute_params):
     return batches
 
 
+def _partition_batches(
+    batches: list[ArrowResultBatch],
+    meta: pd.DataFrame,
+    npartitions: None | int = None,
+    partition_size: None | str | int = None,
+) -> list[list[ArrowResultBatch]]:
+    """
+    Given a list of batches and a sample, partition the batches into dask dataframe
+    partitions.
+
+    Batch sizing is seemingly not under our control, and is typically much smaller
+    than the optimal partition size:
+    https://docs.snowflake.com/en/user-guide/python-connector-distributed-fetch.html
+    So instead batch the batches into partitions of approximately the right size.
+    """
+    if npartitions is not None:
+        assert npartitions >= 1
+        target = sum([b.rowcount for b in batches]) // npartitions
+    elif partition_size is not None:
+        partition_bytes = (
+            parse_bytes(partition_size)
+            if isinstance(partition_size, str)
+            else partition_size
+        )
+        approx_row_size = meta.memory_usage().sum() / len(meta)
+        target = max(partition_bytes / approx_row_size, 1)
+    else:
+        raise ValueError("Must provide one of `npartitions` or `partition_size`")
```

Comment:
I propose that we err if the user provides both.  Thoughts?

Reply:
I originally had that, but had to change it when we changed the default for `partition_size` to `100 MiB`. I don't want to force users to explicitly change that to `None` if they want the `npartitions` branch.

Reply:
Maybe set both as None by default?

```python
if partition_size is None and npartitions is None:
    partition_size = "100 MiB"
```

But then I guess it's hard to get one partition per chunk.  OK.  Happy to skip.

Reply:
Ah, sure, I could go that route. I'm happy to never try to get one partition per chunk, to be honest. I don't think the default chunking is very useful for us today